cpp/include/Ice/ValueFactory.h

```diff
@@ -85,7 +85,7 @@ class ICE_CLASS(ICE_API) ValueFactoryManager
      * @param factory The factory to add.
      * @param id The type id for which the factory can create instances, or an empty string for the default factory.
      */
-    virtual void add(ValueFactoryFunc factory, const ::std::string& id) = 0;
+    virtual void add(ValueFactoryFunc factory, std::string_view id) = 0;
```

Comment:
In theory, it could be / should be a std::string parameter, since the factory typically wants to store this string. However, this results in an inconvenient API for the caller since the caller typically uses a ice_staticId() when calling add. Hence, a string_view parameter.